File: core/src/main/java/info/nightscout/androidaps/interfaces/ActivePluginProvider.java
package info.nightscout.androidaps.interfaces;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public interface ActivePluginProvider {

    @NotNull BgSourceInterface getActiveBgSource(); // Forced to Dexcom

    @NotNull ProfileInterface getActiveProfileInterface(); // Forced to LocalProfile if not changed

    @NotNull InsulinInterface getActiveInsulin(); // Forced to RapidActing if not changed

    @NotNull APSInterface getActiveAPS(); // Forced to SMB

    @NotNull PumpInterface getActivePump(); // Use in places not reachable without active pump. Otherwise IllegalStateException is thrown

    @NotNull SensitivityInterface getActiveSensitivity(); // Forced to oref1 if not changed

    @NotNull TreatmentsInterface getActiveTreatments(); // Forced to treatments

    @NotNull ArrayList<PluginBase> getPluginsList();

    @NotNull ArrayList<PluginBase> getSpecificPluginsVisibleInListByInterface(Class interfaceClass, PluginType type);

    @NotNull ArrayList<PluginBase> getSpecificPluginsVisibleInList(PluginType type);

    @NotNull ArrayList<PluginBase> getSpecificPluginsListByInterface(Class interfaceClass);

//    @NotNull ArrayList<PluginBase> getSpecificPluginsVisibleInList(Class interfaceClass);

    void verifySelectionInCategories();
}

File: core/src/main/java/info/nightscout/androidaps/interfaces/ActivePluginProvider.kt
package info.nightscout.androidaps.interfaces

import java.util.*

interface ActivePluginProvider {

    val activeBgSource: BgSourceInterface // Forced to Dexcom
    val activeProfileInterface: ProfileInterface // Forced to LocalProfile if not changed
    val activeInsulin: InsulinInterface // Forced to RapidActing if not changed
    val activeAPS: APSInterface // Forced to SMB
    val activePump: PumpInterface // Use in places not reachable without active pump. Otherwise IllegalStateException is thrown
    val activeSensitivity: SensitivityInterface // Forced to oref1 if not changed
    val activeTreatments: TreatmentsInterface // Forced to treatments
    val activeOverview: OverviewInterface // Forced to overview

    fun getPluginsList(): ArrayList<PluginBase>

    fun getSpecificPluginsVisibleInListByInterface(interfaceClass: Class<*>, type: PluginType): ArrayList<PluginBase>
    fun getSpecificPluginsVisibleInList(type: PluginType): ArrayList<PluginBase>
    fun getSpecificPluginsListByInterface(interfaceClass: Class<*>): ArrayList<PluginBase>
    fun verifySelectionInCategories()
}